apps/user_ldap/lib/Connection.php

```diff
@@ -594,12 +594,18 @@ private function establishConnection() {
 
 			$isOverrideMainServer = ($this->configuration->ldapOverrideMainServer
 				|| $this->getFromCache('overrideMainServer'));
-			$isBackupHost = (trim($this->configuration->ldapBackupHost) !== "");
+			$isBackupHost = (trim($this->configuration->ldapBackupHost) !== "")
+				&& (!\OC::$CLI || !$this->configuration->ldapBackgroundHost);
```

Comment:
Hard to read, and adds conditions unrelated to the backup host.
I would split into another variable and make the config check more explicit:

```suggestion
				$shouldUseBackgroundHost = \OC::$CLI && (trim($this->configuration->ldapBackgroundHost) !== "");
```